tracer.go

```diff
@@ -73,6 +73,12 @@ type Options struct {
 	// When set, it attempts to exacerbate issues emanating from use of Spans
 	// after calling Finish by running additional assertions.
 	DebugAssertUseAfterFinish bool
+	// DisableSpanPool disables reuse of the underlying span structs.
+	// Buyer Beware, any operations called on a span after Finish is undefined
+	// behavior. The main usecase of disabling the pool is to allow the creation
```

Comment:
> SpanPool were off by default, and a new option 'EnableSpanPool' had to be set to use it

I think that disabling it explicitly is the right first step before explicitly enabling it. This is more of a fix for the time being, and I don't want to break anyone's existing code seeing that lightstep isn't the only project using basictracer-go

> DebugAssertUseAfterFinish disappears (it's implied by EnableSpanPool)

Not necessarily. There's a small window where the span can be put back in the pool but still be usable while it waits to be pulled out of it and reset. Reseting it right when you call finish causes the span to be unusable immediately.